system.h

```diff
@@ -86,20 +90,7 @@ char * stpncpy(char * dest, const char * src, size_t n);
 #define xstrdup(_str) rstrdup((_str))
 #define _free(_ptr) rfree((_ptr))
 
-/* Retrofit glibc __progname */
-#if defined __GLIBC__ && __GLIBC__ >= 2
-#if __GLIBC_MINOR__ >= 1
-#define	__progname	__assert_program_name
-#endif
-#define	setprogname(pn)
-#else
-#define	__progname	program_name
-#define	setprogname(pn)	\
-  { if ((__progname = strrchr(pn, '/')) != NULL) __progname++; \
-    else __progname = pn;		\
-  }
-#endif
-extern const char *__progname;
+#define __progname getprogname()
```

Comment:
There was [a patch in 2013 to fix this issue](http://lists.rpm.org/pipermail/rpm-maint/2013-August/003585.html), but it doesn't look like it was applied. You might want to take that and refactor it for current git master. The current way you're handling the `__progname` stuff is likely not satisfactory.